sdk/core/core-amqp/test/requestResponse.spec.ts

```diff
@@ -174,84 +160,4 @@ describe("RequestResponseLink", function() {
       assert.equal(expectedErrorRegex.test(err.message), true);
     }
   });
-
-  it("should abort a request and response correctly when retried", async function() {
-    const connectionStub = stub(new Connection());
-    const rcvr = new EventEmitter();
-    let messageId: string = "";
-    let counter = 0;
-    connectionStub.createSession.resolves({
-      connection: {
-        id: "connection-1"
-      },
-      createSender: () => {
-        return Promise.resolve({
-          send: (request: any) => {
-            counter++;
-            if (counter != 1) {
-              assert.notEqual(messageId, undefined);
-              assert.notEqual(request.message_id, undefined);
-              assert.notEqual(messageId, request.message_id);
-            }
-            messageId = request.message_id;
-          }
-        });
-      },
-      createReceiver: () => {
-        return Promise.resolve(rcvr);
-      }
-    });
-    const sessionStub = await connectionStub.createSession();
-    const senderStub = await sessionStub.createSender();
-    const receiverStub = await sessionStub.createReceiver();
-    const link = new RequestResponseLink(sessionStub as any, senderStub, receiverStub);
-    const request: AmqpMessage = {
-      body: "Hello World!!"
-    };
-    setTimeout(() => {
-      rcvr.emit("message", {
-        message: {
-          correlation_id: messageId,
-          application_properties: {
-            statusCode: 500,
-            errorCondition: ErrorNameConditionMapper.InternalServerError,
-            statusDescription: "Please retry later.",
-            "com.microsoft:tracking-id": "1"
-          }
-        }
-      });
-    }, 2000);
-    setTimeout(() => {
-      rcvr.emit("message", {
-        message: {
-          correlation_id: messageId,
-          application_properties: {
-            statusCode: 200,
-            errorCondition: null,
-            statusDescription: null,
-            "com.microsoft:tracking-id": null
-          },
-          body: "Hello World!!"
-        }
-      });
-    }, 4000);
-    try {
-      const controller = new AbortController();
-      const signal = controller.signal;
-      setTimeout(controller.abort.bind(controller), 100);
```

Comment:
@chradek, @ShivangiReja I don't recall which of you both added this test case, so mentioning both of you here.

If the intention was to have the abort signal fire in between requests, then shouldn't timeout value here be in between 2000 and 4000 instead of 100?

Reply:
Looks like I was the culprit! You’re absolutely right, this should be something like 3000, not 100.

Reply:
Great, so the intention was to have the abort signal fire between retries.

@ramya0820, I would suggest that we bring this test case back, update the timeout value to 3000, use the approach mentioned in https://github.com/Azure/azure-sdk-for-js/pull/4322#discussion_r305201443 to make the retry work

Reply:
Oh, should this be added to `retry.spec.ts` or in SDK? Because retries are no longer happening on sendRequest here right?

Reply:
That makes sense. We should add tests to `retry.spec.ts` to deal with abort signals. @chradek Can you create an issue for the cases that we would like to cover?

Reply:
I don't think it's really useful to test the `abortSignal` with `retry` since `retry` doesn't accept an `abortSignal` itself. We're using the `abortSignal` to cancel the operation that's passed to `retry`, so really we'd just be testing if `AbortError` was retryable or not.

The existing test would have been more useful before these changes because it makes sure that the operation that calls `retry` is cancelled correctly.

That said... I can open up an issue to look at updating `retry` to accept an abortSignal!

Reply:
#4369 created